include/rtc.h

```diff
@@ -39,12 +122,126 @@ typedef struct rtc_time_t
 extern "C" {
 #endif
 
+/**
+ * @brief High-level convenience helper to initialize the RTC subsystem.
+ * 
+ * The RTC Subsystem depends on the libdragon Timer Subsystem, so make sure
+ * to call #timer_init before calling #rtc_init!
+ *
+ * Some flash carts require the RTC to be explicitly enabled before loading
+ * the ROM file. Some emulators and flash carts do not support RTC at all.
+ *
+ * This function will detect if the RTC is available and if so, will
+ * prepare the RTC so that the current time can be read from it.
+ *
+ * This operation may take up to 50 milliseconds to complete.
+ *
+ * This will also hook the RTC into the newlib gettimeofday function, so
+ * you will be able to use the ISO C time functions if RTC is available.
+ *
+ * @return whether the RTC is present and supported by the RTC Subsystem.
+ */
 bool rtc_init( void );
+
+/**
+ * @brief Close the RTC Subsystem, disabling system hooks.
+ * 
+ * Unhooks the RTC from the newlib gettimeofday function.
+ * This will cause subsequent calls to gettimeofday to error with ENOSYS.
+ */
 void rtc_close( void );
+
+bool rtc_is_source_available( rtc_source_t source );
+
+bool rtc_get_source( void );
+
+/** @brief Switch the preferred source clock for the subsystem.
+ * 
+ * By default, the subsytem will use to the first available source,
+ * but some games may wish to specify the preferred RTC source.
+ * 
+ * Make sure you call #rtc_resync_time after switching sources!
+*/
+bool rtc_set_source( rtc_source_t source );
+
+/**
+ * @brief Resynchronize the subsystem's time with the source clock.
+ * 
+ * You should only need to do this after switching sources.
+ */
+bool rtc_resync_time( void );
+
+time_t rtc_get_time( void );
+
+bool rtc_set_time( time_t new_time );
+
+/**
+ * @brief Determine whether the RTC actually supports writing the time.
+ *
+ * Some emulators and flash carts do not support writing to the RTC, so
+ * this function makes an attempt to detect silent write failures and will
+ * return `false` if it is unable to change the time on the RTC.
+ *
+ * This function is useful if your program wants to conditionally offer the
+ * ability to set the time based on hardware/emulator support.
+ *
+ * Unfortunately this operation may introduce a slight drift in the clock,
+ * but it is the only way to determine if the RTC supports the write command.
+ *
+ * This operation will take approximately 1 second to complete.
+ *
+ * @return whether RTC write persistence appears to be supported
+ */
+bool rtc_is_persistent( void );
+
+__attribute__((deprecated("use rtc_is_persistent instead")))
 bool rtc_is_writable( void );
+
+/**
+ * @brief Read the current date/time from the real-time clock.
+ *
```

Comment:
```suggestion
 * @deprecated Use rtc_get_time() instead.
 
```
Should you also mark it deprecated for the Doxygen?